.github/workflows/docker.yml

```diff
@@ -18,68 +22,78 @@ concurrency:
   group: ${{ github.workflow }}-${{ github.ref || github.run_id }}
   cancel-in-progress: true
 jobs:
-  docker:
+  heighliner-docker:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
+
+      - uses: docker/setup-buildx-action@v3
+
+      - uses: docker/setup-qemu-action@v2
+
+      - uses: docker/metadata-action@v5
+        id: meta
+        with:
+          images: ${{ env.DOCKER_IMAGE }}
+          flavor: |
+            prefix=heighliner-,onlatest=true
+          tags: |
+            type=edge
+            type=ref,enable=${{ github.event_name == 'pull_request' }},event=branch
+            type=ref,event=tag
+
```

Comment:
`DOCKER_IMAGE` is an environment variable (sting) defined above. Secrets are significantly less flexible because they're harder to change, and change one affects all branches. So you can't change it and test it before merging it. Secrets are also significantly less maintainable because there's no way to find out what their current value is.

What is wrong with you?